packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx

```diff
@@ -39,17 +41,28 @@ const Container = styled.div<{ $variant?: Exclude<ButtonVariant, 'danger'> }>`
 `;
 
 const checkChildren = (children: Array<React.ReactNode>) =>
-    children.every(
-        child =>
-            React.isValidElement(child) && (child.type === Button || child.type === IconButton),
-    );
+    children.every(child => {
+        if (React.isValidElement(child)) {
+            if (child.type === Button || child.type === IconButton) {
```

Comment:
nit: does `[Button, IconButton].includes(child.type)` works. It yes, we can put them in array `const  ALLOWED_NODES = [...]`

Reply:
it doesn't work here

```
Argument of type 'string | JSXElementConstructor<any>' is not assignable to parameter of type '(({ variant, size, isDisabled, isLoading, isFullWidth, icon, iconSize, iconAlignment, type, children, margin, ...rest }: ButtonProps) => Element) | (({ icon, label, bottomLabel, variant, size, iconSize, isDisabled, isLoading, ...rest }: IconButtonProps) => Element)'.
  Type 'string' is not assignable to type '(({ variant, size, isDisabled, isLoading, isFullWidth, icon, iconSize, iconAlignment, type, children, margin, ...rest }: ButtonProps) => Element) | (({ icon, label, bottomLabel, variant, size, iconSize, isDisabled, isLoading, ...rest }: IconButtonProps) => Element)'.ts(2345)
```

Reply:
:(